src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -648,6 +647,53 @@ public byte[] Http2AltSvcOriginUri
             return await GetHttpConnectionAsync(request, async, cancellationToken).ConfigureAwait(false);
         }
 
+        private Http2Connection? GetExistingHttp2Connection()
+        {
+            Http2Connection[]? localConnections = _http2Connections;
```

Comment:
It's probably worth a comment here to explain that the array itself is never mutated, though the array reference may change.

Reply:
It's [explained](https://github.com/dotnet/runtime/blob/8af737a3c92a6ae10af20e3a1ea2ebb4a0dc04f1/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs#L69) above `_http2Connections` declaration.